.github/workflows/ci.yml

```
@@ -19,12 +19,52 @@ jobs:
         os: [ubuntu-latest]
         ghc: ['8.0', '8.2', '8.4', '8.6', '8.8', '8.10', '9.0', '9.2']
         include:
-        - os: windows-latest
-          ghc: 'latest'
         - os: macOS-latest
           ghc: 'latest'
     steps:
     - uses: actions/checkout@v2
+    - uses: haskell/actions/setup@v1
+      id: setup-haskell-cabal
+      with:
+        ghc-version: ${{ matrix.ghc }}
+    - name: Update cabal package database
+      run: cabal update
+    - uses: actions/cache@v2
+      name: Cache cabal stuff
+      with:
+        path: |
+          ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
+          dist-newstyle
+        key: ${{ runner.os }}-${{ matrix.ghc }}
+    - name: Test
+      run: |
+        cabal sdist -z -o .
+        cabal get bytestring-*.tar.gz
+        cd bytestring-*/
+        cabal build bytestring:tests --enable-tests --enable-benchmarks
+        cabal test --enable-tests --enable-benchmarks --test-show-details=direct all -j1
```

```suggestion
        cabal test --enable-tests --enable-benchmarks --test-show-details=direct all
```
`-j1` was an attempt to stabilise Windows build.